src/basic/fs-util.c

```diff
@@ -741,18 +741,27 @@ bool unsafe_transition(const struct stat *a, const struct stat *b) {
         return a->st_uid != b->st_uid; /* Otherwise we need to stay within the same UID */
 }
 
-static int log_unsafe_transition(int a, int b, const char *path, unsigned flags) {
-        _cleanup_free_ char *n1 = NULL, *n2 = NULL;
+static int log_unsafe_transition(int a, int b, unsigned flags) {
+        _cleanup_free_ char *n1 = NULL, *n2 = NULL, *user_a = NULL, *user_b = NULL;
+        struct stat stat_a, stat_b;
 
         if (!FLAGS_SET(flags, CHASE_WARN))
                 return -ENOLINK;
 
         (void) fd_get_path(a, &n1);
         (void) fd_get_path(b, &n2);
 
+        if (fstat(a, &stat_a) < 0)
+                return -errno;
+        if (fstat(b, &stat_b) < 0)
+                return -errno;
+
+        user_a = uid_to_name(stat_a.st_uid);
+        user_b = uid_to_name(stat_b.st_uid);
+
         return log_warning_errno(SYNTHETIC_ERRNO(ENOLINK),
-                                 "Detected unsafe path transition %s %s %s during canonicalization of %s.",
-                                 strna(n1), special_glyph(SPECIAL_GLYPH_ARROW), strna(n2), path);
+                                 "Path transition %s %s %s is unsafe: owner is %s, wanted %s",
```

Comment:
Hm, I'd like to keep `path` in the log. How about the following?
```
"Detected unsafe path transition %s (owned by %s) %s %s (owned by %s) during canonicalization of %s.",
strna(n1), strna(user_a), special_glyph(SPECIAL_GLYPH_ARROW), strna(n2), strna(user_b), path
```

Also, as you can see in this suggestion, please use `strna()` for `user_a` and `user_b`, as they may be NULL.

Reply:
Makes sense to me